### PR TITLE
More Metrics internal logging

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes #.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by cijothomas** [comment]
related to https://github.com/open-telemetry/opentelemetry-dotnet/issues/2621

File: src/OpenTelemetry/Metrics/MetricReaderExt.cs [hunk 1]
```diff
@@ -104,25 +104,20 @@ internal List<Metric> AddMetricsListWithViews(Instrument instrument, List<Metric
...
+                        OpenTelemetrySdkEventSource.Log.MetricInstrumentIgnored(metricName, instrument.Meter.Name, "View configuration asks to drop this instrument.", "Modifiy view configuration to allow this instrument, if desired.");
                         continue;
                     }
 
                     var index = ++this.metricIndex;
                     if (index >= this.maxMetricStreams)
                     {
-                        // TODO: Log that instrument is ignored
-                        // as max number of Metrics have reached.
+                        OpenTelemetrySdkEventSource.Log.MetricInstrumentIgnored(metricName, instrument.Meter.Name, "Maximum allowed Metric streams for the provider exceeded.", "Use views to drop unused instruments. Or use SetMaxMetricStreams to configure Provider to allow higher limit.");
```
Comment (reyang) [inline src/OpenTelemetry/Metrics/MetricReaderExt.cs:120]:
```suggestion
                        OpenTelemetrySdkEventSource.Log.MetricInstrumentIgnored(metricName, instrument.Meter.Name, "Maximum allowed Metric streams for the provider exceeded.", "Use MeterProvider.AddView to drop unused instruments. Or use SetMaxMetricStreams to configure Provider to allow higher limit.");
```

File: src/OpenTelemetry/Metrics/MetricReaderExt.cs [hunk 2]
```diff
@@ -104,25 +104,20 @@ internal List<Metric> AddMetricsListWithViews(Instrument instrument, List<Metric
...
+                        OpenTelemetrySdkEventSource.Log.MetricInstrumentIgnored(metricName, instrument.Meter.Name, "Metric name conflicting with existing name.", "Either change the name of the instrument or change name using View.");
                         continue;
                     }
 
                     if (metricStreamConfig?.Aggregation == Aggregation.Drop)
                     {
-                        // TODO: Log that instrument is ignored
-                        // as user explicitly asked to drop it
-                        // with View.
+                        OpenTelemetrySdkEventSource.Log.MetricInstrumentIgnored(metricName, instrument.Meter.Name, "View configuration asks to drop this instrument.", "Modifiy view configuration to allow this instrument, if desired.");
```
Comment (reyang) [inline src/OpenTelemetry/Metrics/MetricReaderExt.cs:113]:
```suggestion
                        OpenTelemetrySdkEventSource.Log.MetricInstrumentIgnored(metricName, instrument.Meter.Name, "View configuration asks to drop this instrument.", "Modify View configuration to allow this instrument, if desired.");
```

File: src/OpenTelemetry/Metrics/MeterProviderSdk.cs [hunk 3]
```diff
@@ -113,49 +113,56 @@ internal sealed class MeterProviderSdk : MeterProvider
...
-                    // GC right after this method.
-                    var metricStreamConfigs = new List<MetricStreamConfiguration>(viewConfigCount);
-                    foreach (var viewConfig in this.viewConfigs)
+                    try
                     {
-                        var metricStreamConfig = viewConfig(instrument);
-                        if (metricStreamConfig != null)
+                        // Creating list with initial capacity as the maximum
+                        // possible size, to avoid any array resize/copy internally.
+                        // There may be excess space wasted, but it'll eligible for
```
Comment (reyang) [inline src/OpenTelemetry/Metrics/MeterProviderSdk.cs:120]: This doesn't seem to be the correct English.

Comment (cijothomas) [reply src/OpenTelemetry/Metrics/MeterProviderSdk.cs:120]: yes. This was not added in this PR. There are more cleanups coming and will fix

**Review by reyang — APPROVED** [review]

**Pull request merged**